src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNICommon.cs

```diff
@@ -99,6 +102,44 @@ internal enum SNISMUXFlags
         SMUX_DATA = 8       // SMUX data packet
     }
 
+    internal class SslStreamProxy : SslStream
+    {
+        private Task _currentTask;
+
+        #region constructors
+        public SslStreamProxy(Stream innerStream) : base(innerStream)
+        {
+        }
+
+        public SslStreamProxy(Stream innerStream, bool leaveInnerStreamOpen) : base(innerStream, leaveInnerStreamOpen)
+        {
+        }
+
+        public SslStreamProxy(Stream innerStream, bool leaveInnerStreamOpen, RemoteCertificateValidationCallback userCertificateValidationCallback) : base(innerStream, leaveInnerStreamOpen, userCertificateValidationCallback)
+        {
+        }
+
+        public SslStreamProxy(Stream innerStream, bool leaveInnerStreamOpen, RemoteCertificateValidationCallback userCertificateValidationCallback, LocalCertificateSelectionCallback userCertificateSelectionCallback) : base(innerStream, leaveInnerStreamOpen, userCertificateValidationCallback, userCertificateSelectionCallback)
+        {
+        }
+
+        public SslStreamProxy(Stream innerStream, bool leaveInnerStreamOpen, RemoteCertificateValidationCallback userCertificateValidationCallback, LocalCertificateSelectionCallback userCertificateSelectionCallback, EncryptionPolicy encryptionPolicy) : base(innerStream, leaveInnerStreamOpen, userCertificateValidationCallback, userCertificateSelectionCallback, encryptionPolicy)
+        {
+        }
+        #endregion
```

Comment:
You don't need all the constructors actually..

```suggestion
        public SslStreamProxy(Stream innerStream, bool leaveInnerStreamOpen, RemoteCertificateValidationCallback userCertificateValidationCallback) 
            : base(innerStream, leaveInnerStreamOpen, userCertificateValidationCallback)
        { }
```